ParProcCo/job_scheduler.py

```diff
@@ -418,82 +587,131 @@ def _report_job_info(self) -> None:
+                job_scheduling_info.set_completion_status(True)
                 status_info.final_state = SLURMSTATE.COMPLETED
-                if status_info.slots and status_info.wall_time:
-                    cpu_time = timedelta(
-                        seconds=float(status_info.wall_time * status_info.slots)
-                    )
+                if status_info.cpus and status_info.wall_time:
+                    cpu_time = str(status_info.wall_time * status_info.cpus)
                 else:
                     cpu_time = "n/a"
                 logging.info(
-                    f"Job {job_id} with args {self.jobscript_args} completed."
-                    f" CPU time: {cpu_time}; Slots: {status_info.slots}"
-                    f" Dispatch time: {status_info.time_to_dispatch}; Wall time: {status_info.wall_time}."
+                    f"Job {job_id} with args {job_scheduling_info.job_script_arguments}"
+                    f" completed. CPU time: {cpu_time}; Slots: {status_info.cpus}"
+                    f" Dispatch time: {status_info.time_to_dispatch};"
+                    f" Wall time: {status_info.wall_time}."
                 )
             else:
                 status_info.final_state = state
                 logging.error(
                     f"Job {job_id} ended with job state {status_info.final_state}"
-                    f" Args {self.jobscript_args};"
-                    f" Dispatch time: {status_info.time_to_dispatch}; Wall time: {status_info.wall_time}."
+                    f" Args {job_scheduling_info.job_script_arguments};"
+                    f" Dispatch time: {status_info.time_to_dispatch};"
+                    f" Wall time: {status_info.wall_time}."
                 )
 
-            self.job_history[self.batch_number][job_id] = status_info
+        self.job_history.append({jsi.job_id: jsi for jsi in job_scheduling_info_list})
 
-    def resubmit_jobs(self, job_indices: List[int]) -> bool:
-        self.batch_number += 1
-        self.job_history[self.batch_number] = {}
-        self.job_completion_status = {str(i): False for i in job_indices}
-        logging.info(f"Resubmitting jobs with job_indices: {job_indices}")
-        return self._submit_and_monitor(job_indices)
-
-    def filter_killed_jobs(self, jobs: List[StatusInfo]) -> List[int]:
+    def resubmit_jobs(
+        self, job_ids: list[int] | None = None, batch: int | None = None
+    ) -> bool:
+        old_job_scheduling_info_dict = self.get_job_history_batch(batch_number=batch)
+        new_job_scheduling_info_list = []
+        for job_id, old_job_scheduling_info in old_job_scheduling_info_dict.items():
+            if job_ids is None or job_id in job_ids:
+                new_job_scheduling_info = deepcopy(old_job_scheduling_info)
+                new_job_scheduling_info.set_completion_status(False)
+                new_job_scheduling_info.status_info = None
+                new_job_scheduling_info.job_id = -1
```

Comment:
Changing this will affect checks like line 157. Is -1 what is returned by the scheduler if it fails to be assigned properly?

Reply:
I didn't want it to be optional